tests/linearGaussian_vector_field_test.py

```diff
@@ -0,0 +1,585 @@
+    # (otherwise the posterior gets too tight and the c2st is too high),
+    # but it doesn't really improve the results for both FMPE and NPSE.
+    likelihood_cov = 0.9 * eye(num_dim)
+
+    if prior_type == "gaussian" or (prior_type is None):
+        prior_mean = zeros(num_dim)
+        prior_cov = eye(num_dim)
+        prior = MultivariateNormal(loc=prior_mean, covariance_matrix=prior_cov)
+        prior_npse = prior if prior_type is None else None
+    elif prior_type == "uniform":
+        prior = BoxUniform(-2 * ones(num_dim), 2 * ones(num_dim))
+        prior_npse = prior
+
+    # This check that our method to handle "general" priors works.
+    # i.e. if NPSE does not get a proper passed by the user.
+    if vector_field_type == "fmpe":
+        inference = FMPE(prior_npse, show_progress_bars=True)
+    else:
+        inference = NPSE(
+            prior_npse, show_progress_bars=True, sde_type=vector_field_type
+        )
+
+    theta = prior.sample((num_simulations,))
+    x = linear_gaussian(theta, likelihood_shift, likelihood_cov)
+
+    score_estimator = inference.append_simulations(theta, x).train(
+        stop_after_epochs=200,
+        training_batch_size=100,
+        max_num_epochs=50,
+    )
+
+    return {
+        "score_estimator": score_estimator,
+        "inference": inference,
+        "prior": prior,
+        "likelihood_shift": likelihood_shift,
+        "likelihood_cov": likelihood_cov,
+        "prior_mean": prior_mean
+        if prior_type == "gaussian" or prior_type is None
+        else None,
+        "prior_cov": prior_cov
+        if prior_type == "gaussian" or prior_type is None
+        else None,
+        "num_dim": num_dim,
+        "vector_field_type": vector_field_type,
+    }
+
+
+@pytest.mark.slow
+def test_vector_field_sde_ode_sampling_equivalence(vector_field_trained_model):
+    """
+    Test whether SDE and ODE sampling are equivalent
+    for FMPE and NPSE.
+    """
+    num_samples = 1000
+    x_o = zeros(1, vector_field_trained_model["num_dim"])
+
+    inference = vector_field_trained_model["inference"]
+    vector_field_type = vector_field_trained_model["vector_field_type"]
+    sde_posterior = inference.build_posterior(sample_with="sde").set_default_x(x_o)
+    ode_posterior = inference.build_posterior(sample_with="ode").set_default_x(x_o)
+
+    sde_samples = sde_posterior.sample((num_samples,))
+    ode_samples = ode_posterior.sample((num_samples,))
+
+    check_c2st(
+        sde_samples,
+        ode_samples,
+        alg=f"sample_methods_equivalence-{vector_field_type}",
+        tol=0.07,
+    )
+
+
+# TODO: Currently, c2st is too high for FMPE (e.g., > 3 number of observations),
+# so some tests are skipped so far. It is unclear at the moment whether this is due to
+# some issues with the implementation, the nn architecture, or it is simply the case
+# with flow matching (compositional flow matching hasn't been tested yet elsewhere).
```

Comment:
can you please still add them explicitly as test cases here, add a `skip`, and create an issue, thanks! 🙏